crates/sql_provider_datafusion/src/dbconnectionpool.rs

```diff
@@ -0,0 +1,41 @@
+use std::{collections::HashMap, sync::Arc};
+
+use snafu::prelude::*;
+use spicepod::component::dataset::acceleration;
+
+use crate::dbconnection::DbConnection;
+
+pub mod duckdbpool;
+
+#[derive(Debug, Snafu)]
+pub enum Error {
+    #[snafu(display("DuckDBError: {source}"))]
+    DuckDBError { source: duckdb::Error },
```

Comment:
I think we have a tendency not exposing implementation details on the trait level

Reply:
no I am wrong.